app/models/incomeSourceDetails/IncomeSourceDetailsResponseModel.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.incomeSourceDetails
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.{Json, Reads, _}
+
+import java.time.LocalDate
+
+sealed trait IncomeSourceDetailsResponseModel
+
+case class IncomeSourceDetailsModel(nino: String,
+                                    mtdbsa: String,
+                                    yearOfMigration: Option[String],
+                                    businesses: List[BusinessDetailsModel],
+                                    property: Option[PropertyDetailsModel]) extends IncomeSourceDetailsResponseModel {
+
+  def taxYears: Seq[Int] = Option(orderedTaxYearsByAccountingPeriods).filter(_.nonEmpty).getOrElse(Seq(getCurrentTaxEndYear))
+
+  def startingTaxYear: Int = (businesses.flatMap(_.firstAccountingPeriodEndDate) ++ property.flatMap(_.firstAccountingPeriodEndDate))
+    .map(_.getYear).sortWith(_ < _).headOption.getOrElse(throw new RuntimeException("User missing first accounting period information"))
+
+  def orderedTaxYearsByAccountingPeriods: List[Int] = {
+    (startingTaxYear to getCurrentTaxEndYear).toList
```

Comment:
should add a .distinct to the list so if the startingTaxYear and getCurrentTaxEndYear are the same they don't appear in the list twice.

Reply:
👍